### DISCUSSION TITLE
Form Filling

### DISCUSSION THREAD

**Discussion by hartoflearning** [speechly/speechly#200] [Q&A] [answered]
I'm having a little trouble getting the speechly text to fill out the forms when I worked through the Lesson 5/5 Tutorial on the website. Perhaps, it's some small issue like I'm not connecting the javascript file to the html file correctly.
Both files are stored in the same folder (on my D drive).
I put in the api code and can get the microphone input and I can tell that is configured correctly, because it highlights the shipping and address types in blue. (I love that visual feedback by the way~ the thing that makes Speechly unique and super cool). 
However, it just does not fill out the forms as it shows the demo.
Note: I am not putting the api on the code I am loading here, but I did on the code on my computer.  I did not put in a capture key on my computer code- should I have? Where would I get that info if needed?
Here is the code in the index.html file:
`
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <script type="text/javascript" src="https://speechly.github.io/browser-ui/v3/push-to-talk-button.js"></script>
    <script type="text/javascript" src="https://speechly.github.io/browser-ui/v3/big-transcript.js"></script>
    <link rel-"stylesheet" href="https://speechly.github.io/browser-ui/v3/speechly-ui.css">
    <title>Document</title>
</head>

<body>
    <script src="main.js"></script>
    <div class="PushToTalkContainer">
        <push-to-talk-button appid="" capturekey=" " />
    </div>
    <div class="BigTranscriptContainer">
        <big-transcript></big-transcript>
    </div>
    <main>
        <section>
            <div class="group">
                <label>Shipping</label>
                <div class="InputContaincer">
                    <select id="Shipping">
                        <option>STANDARD</option>
                        <option>EXPEDITED</option>
                        <option>PRIORITY</option>
                    </select>
                </div>
            </div>
            <div class="group">
                <label>Steet Address</label>
                <div class="InputContaincer">
                    <input id="street_address" type-"text" placeholder-="" />
                </div>
            </div>
        </section>
    </main>
</body>

</html>
`
Here is the code from the main.js file:
`
document
    .getElementsByTagName("push-to-talk-button")[0]
    .addEventListener("speechsegment", (e) => {
        e.detail.entities.forEach(entity => {
            document.getElementById(entity.type).value = entity.value;
        })
    });
`

**Comment by thisismatu** [comment, marked as answer]
Hi @hartoflearning 👋 

Try importing the Speechly browser-ui components from a CDN like unpkg instead. Learn more about [client libraries from our docs](https://docs.speechly.com/client-libraries/usage/).

```
<script type="text/javascript" src="https://unpkg.com/@speechly/browser-ui/core/push-to-talk-button.js"></script>
<script type="text/javascript" src="https://unpkg.com/@speechly/browser-ui/core/big-transcript.js"></script>
```

Also, don't forget to add the `App ID` for your deployed application to to the push talk button. You can obtain the App ID from the [Dashboard](https://api.speechly.com/dashboard/#/dashboard).

```
<push-to-talk-button appid="your-app-id-here" capturekey=" " />
```

Finally, make sure that your input IDs in your HTML code match your Entities. Since this particular application is using those as identifiers.

![Untitled](https://user-images.githubusercontent.com/2579244/172316981-9064fae3-88f5-4def-a503-ecb31aac996e.png)

By doing these changes i got it up and running. Happy developing!
